lingua_franca/lang/common_data_fa.py

```diff
@@ -0,0 +1,230 @@
+#
+# Copyright 2017 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from collections import OrderedDict
+from .parse_common import invert_dict
+
+_FUNCTION_NOT_IMPLEMENTED_WARNING = "تابع خواسته شده در زبان فارسی پیاده سازی نشده است."
+
+_NUM_STRING_FA = {
+    0: 'صفر',
+    1: 'یک',
+    2: 'دو',
+    3: 'سه',
+    4: 'چهار',
+    5: 'پنج',
+    6: 'شش',
+    7: 'هفت',
+    8: 'هشت',
+    9: 'نه',
+    10: 'ده',
+    11: 'یازده',
+    12: 'دوازده',
+    13: 'سیزده',
+    14: 'چهارده',
+    15: 'پونزده',
+    16: 'شونزده',
+    17: 'هیفده',
+    18: 'هیجده',
+    19: 'نوزده',
+    20: 'بیست',
+    30: 'سی',
+    40: 'چهل',
+    50: 'پنجاه',
+    60: 'شصت',
+    70: 'هفتاد',
+    80: 'هشتاد',
+    90: 'نود'
+}
+
+
+_FRACTION_STRING_FA = {
+    2: 'دوم',
+    3: 'سوم',
+    4: 'چهارم',
+    5: 'پنجم',
+    6: 'ششم',
+    7: 'هفتم',
+    8: 'هشتم',
+    9: 'نهم',
+    10: 'دهم',
+    11: 'یازدهم',
+    12: 'دوازدهم',
+    13: 'سیزدهم',
+    14: 'چهاردهم',
+    15: 'پونزدهم',
+    16: 'شونزدهم',
+    17: 'هیفدهم',
+    18: 'هیجدهم',
+    19: 'نوزدهم',
+    20: 'بیستم'
+}
+
+
+_LONG_SCALE_FA = OrderedDict([
+    (100, 'صد'),
+    (1000, 'هزار'),
+    (1000000, 'میلیون'),
+    (1e12, "تریلیارد"),
+])
+
+
+_SHORT_SCALE_FA = OrderedDict([
+    (100, 'صد'),
+    (1000, 'هزار'),
+    (1000000, 'میلیون'),
+    (1e9, "میلیارد"),
+    (1e12, 'تریلیون'),
+    (1e15, "تریلیارد"),
+])
+
+
+_ORDINAL_BASE_FA = {
+    1: 'اول',
+    2: 'دوم',
+    3: 'سوم',
+    4: 'چهارم',
+    5: 'پنجم',
+    6: 'ششم',
+    7: 'هفتم',
+    8: 'هشتم',
+    9: 'نهم',
+    10: 'دهم',
+    11: 'یازدهم',
+    12: 'دوازدهم',
+    13: 'سیزدهم',
+    14: 'چهاردهم',
+    15: 'پونزدهم',
+    16: 'شونزدهم',
+    17: 'هیفدهم',
+    18: 'هیجدهم',
+    19: 'نوزدهم',
+    20: 'بیستم',
+    30: 'سیم',
```

Comment:
these guys shouldn't have been translated in this way, that's an informal/casual way of saying the words, so, although we might encounter the words and numbers in this way more often, the actual representation is `سی‌ام` instead of `سیم‍`. but, there is another `سی ام` thing going on too. (hassles of writing and speaking Persian I guess?)
and, we actually have a translation for `wire` for the word `سیم` too if that's not complicated enough.

so, there is a compromise to be made, or a solution should be provided since there are sometimes up to 3-4 ways of writing the same word and all are true.

Reply:
In parsing I handle (or tried to handle) all cases but in formatting I think we have to choose one of the styles.

Reply:
look at the catalan PR, i added a decorator for a similar use case, theres 4 different ways to say the time in catalan, this is similar

i suppose this could be used to let user select which variant to use, will require some reworking of the internals so maybe leave it for a follow up PR

Reply:
We (read: @JarbasAl) recently introduced a function decorator to help in situations like this, at `internal.lookup_variant()`, but I couldn't possibly say if it would help in this *specific* situation.

Reply:
So I should add a parameter `variant` to format functions with two possible value `formal` (for `پونزدهم`) and `conversational` (for `پانزدهم`) ?
In case (`سیم` vs `سی ام`) I think it does not matter so much in formatting, The pronunciation of both of them are equal and we can choose one of them (I prefer `سی ام`). But `پونزدهم` is important.

Reply:
yep exactly, in whatever methods you think it makes sense you can use that decorator. catalan language is the only usage in the wild so far and is a good template

but keep in mind these are not yet used by mycroft-core, a follow up PR will be needed to actually select those variants depending on user config. But the first step is getting the functionality here ;)